onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -1066,6 +1066,16 @@ DataLayout QNNExecutionProvider::GetPreferredLayout() const {
   return DataLayout::NHWC;
 }
 
+std::optional<bool> QNNExecutionProvider::ShouldConvertNodeLayoutToNhwc(std::string_view node_domain,
+                                                                        std::string_view node_op_type) const {
+  if (node_domain == kOnnxDomain && node_op_type == "Upsample") {
+    // Upsample is translated to QNN's Resize, which requires the NHWC layout for processing.
+    return true;
+  }
```

Comment:
The Op list for QNN EP is not correct. should include Conv, ConvTranspose, Pooling ops, etc. Please double check.

Reply:
Please check the code path for
[https://github.com/microsoft/onnxruntime/blob/a17e3462ffb44df06299bcbc9af7de23f5eba367/onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc#L3304](https://github.com/microsoft/onnxruntime/blob/a17e3462ffb44df06299bcbc9af7de23f5eba367/onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc#L3304)